pkg/resourcemonitor/podresourcesscanner.go

```diff
@@ -57,58 +57,49 @@ func NewPodResourcesScanner(namespace string, podResourceClient podresourcesapi.
 }
 
 // isWatchable tells if the the given namespace should be watched.
-func (resMon *PodResourcesScanner) isWatchable(podNamespace string, podName string, hasDevice bool) (bool, bool, error) {
-	pod, err := resMon.k8sClient.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+// In Scan(), if watchable is false, this pods scan will skip
+// so we can return directly if pod's namespace is not watchable
+func (resMon *PodResourcesScanner) isWatchable(podResource *podresourcesapi.PodResources) (bool, bool, error) {
+	if resMon.namespace != "*" && resMon.namespace != podResource.Namespace {
+		return false, false, nil
+	}
+
+	pod, err := resMon.k8sClient.CoreV1().Pods(podResource.Namespace).Get(context.TODO(), podResource.Name, metav1.GetOptions{})
 	if err != nil {
 		return false, false, err
 	}
 
-	isIntegralGuaranteed := hasExclusiveCPUs(pod)
+	podHasExclusiveCPUs := hasExclusiveCPUs(pod)
```

Comment:
nit: somehow this naming makes me confused. `hasExclusiveCPUs(pod)` tells if any containers within has integral cpu requests but doesn't look at the Pod QoS (iow to be correct `hasExclusiveCPUs` would be `hasExclusiveCPUs(pod) && isPodGuaranteed`)

What do you think about renaming a bit like:

```suggestion
	hasIntegralCPUs := podHasIntegralCPUs(pod)
```

Reply:
Oh, looking at @ffromani's earlier comment (https://github.com/kubernetes-sigs/node-feature-discovery/pull/1979#discussion_r1906778199) another possibility (instead of renaming) would be to change `hasExclusiveCPUs(pod)` to behave like it name suggests. That is, check the pods QoS inside the function. Or does this break something elsewhere (non-Guaranteed pods with integral CPUs)(?) Should not afaiu.

Thoughts @ffromani ?

Reply:
ping @AllenXu93 WDYT?